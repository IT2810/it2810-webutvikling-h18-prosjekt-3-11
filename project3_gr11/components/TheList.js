import React, { Component } from 'react';
import {ListView, TextInput, View} from 'react-native';
import {Button, CheckBox, Icon, List, ListItem, Text } from 'native-base';

export class TheList extends Component {
    constructor(props) {
        super(props);
        this.ds = new ListView.DataSource({ rowHasChanged: (r1, r2) => r1 !== r2 });
        this.state = {
            newTodoInput: "",
            // Array of todos
            listOfTodos: [
                { id: 2, task: "Bli ferdig med denne greia" },
                { id: 4, task: "En oppgave" },
                { id: 6, task: "En annen oppgave som har veldig lang tekst Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor" },
                { id: 8, task: "FullfÃ¸rt oppgave" },
                { id: 10, task: "Jeg hater denne lista" },
            ],
            // Array with checked todos
            selectedTodoId: []
        };
    }

    // Add data from inputfield with date and push it in the todolist
    addNewTodoInput = (txt) => {
        let listOfTodos = this.state.listOfTodos;
        listOfTodos.push({id: new Date(), task: txt});
        // Set the list
        this.setState({listOfTodos})
    }

    // Takes the uncheck or check a item from the list
    onCheckBoxPress(id) {
        let checkBoxState = this.state.selectedTodoId;
        if ( checkBoxState.includes( id ) ) {
            // Delete item from id which unchecks it
            checkBoxState.splice( checkBoxState.indexOf(id), 1 );
        } else {
            // Add item to array which checks it
            checkBoxState.push( id );
        }

        // Set the new array of checked items
        this.setState({
            selectedTodoId: checkBoxState
        });
        console.log(this.state.selectedTodoId);
    }

    // Deletes the selected item from the list
    deleteRow(secId, rowId, rowMap) {
        // Grab reference to this row
        rowMap[`${secId}${rowId}`].props.closeRow();
        const newData = [...this.state.listOfTodos];
        // Shrink the list from the rowsId
        newData.splice(rowId, 1);
        // Set the new array above
        this.setState({ listOfTodos: newData });
        const mordi = {hei: secId, hallo: rowId, fardi: rowMap};
        console.log(mordi.hei);
        console.log(mordi.hallo);
    }
    render() {
        // Checks if  row is changes and set it as true,
        // and set the new defines the new related data to the data source
        // r1 = old, r2 = new (if it's alike then it's true and row is changed)
        const ds = new ListView.DataSource({ rowHasChanged: (r1, r2) => r1 !== r2 });
        return (
            <View>
                <TextInput
                    placeholder='Enter a task...'
                    onChangeText= {(text) => this.setState({task:text})}
                />
                <Button bordered
                        onPress={() =>  this.addNewTodoInput(this.state.task)}>
                    <Text> add </Text>
                </Button>
                <List
                    rightOpenValue={-75}
                    enableEmptySections={true}
                    // Set the datasource to the array with the list of todos
                    dataSource={this.ds.cloneWithRows(this.state.listOfTodos)}
                    renderRow={(listOfTodos) =>
                        <ListItem>
                            <CheckBox
                                /* Set a todo to check if the ID is in the array */
                                checked={this.state.selectedTodoId.includes(listOfTodos.id)}
                                /* Add/delete todo from/to array if it's checked or unchecked */
                                onPress={()=>this.onCheckBoxPress(listOfTodos.id)}
                            />
                            <Text>{listOfTodos.task}</Text>
                        </ListItem>}
                    renderRightHiddenRow={(data, secId, rowId, rowMap) =>
                        /* Takes the references to the selected row and takes it in the function */
                        <Button full danger onPress={_ => this.deleteRow(secId, rowId, rowMap)}>
                            <Icon active name="trash" />
                        </Button>
                    }
                />
            </View>
        );
    }
}