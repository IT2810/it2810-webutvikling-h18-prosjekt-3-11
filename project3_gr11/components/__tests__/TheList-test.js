import React from 'react';
import renderer from 'react-test-renderer';
import { TheList } from "../TheList";

test('renders correctly', () => {
    const tree = renderer.create(<TheList/>).toJSON();
    expect(tree).toMatchSnapshot();
});
