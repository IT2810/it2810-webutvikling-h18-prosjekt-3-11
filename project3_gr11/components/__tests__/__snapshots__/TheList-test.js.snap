// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View>
  <TextInput
    allowFontScaling={true}
    multiline={true}
    onChangeText={[Function]}
    placeholder="Enter a task..."
    style={
      Object {
        "fontSize": 30,
        "marginLeft": 10,
        "marginRight": 10,
        "marginTop": 10,
      }
    }
  />
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "center",
        "backgroundColor": "transparent",
        "borderBottomWidth": 1,
        "borderColor": "#007aff",
        "borderLeftWidth": 1,
        "borderRadius": 5,
        "borderRightWidth": 1,
        "borderTopWidth": 1,
        "borderWidth": 1,
        "elevation": null,
        "flexDirection": "row",
        "height": 60,
        "justifyContent": "space-between",
        "marginBottom": 10,
        "marginTop": 10,
        "opacity": 1,
        "paddingBottom": 6,
        "paddingTop": 6,
        "shadowColor": null,
        "shadowOffset": null,
        "shadowOpacity": null,
        "shadowRadius": null,
        "width": "90%",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "color": "#007aff",
            "fontFamily": "System",
            "fontSize": 16.5,
            "marginLeft": 0,
            "marginRight": 0,
            "paddingLeft": 16,
            "paddingRight": 16,
          },
          Object {
            "marginLeft": "40%",
          },
        ]
      }
      uppercase={false}
    >
       add 
    </Text>
  </View>
  <RCTScrollView
    closeOnRowBeginSwipe={false}
    closeOnRowPress={true}
    closeOnScroll={true}
    dataSource={
      ListViewDataSource {
        "items": 0,
      }
    }
    directionalDistanceChangeThreshold={2}
    disableLeftSwipe={false}
    disableRightSwipe={false}
    enableEmptySections={true}
    leftOpenValue={0}
    onScroll={[Function]}
    previewFirstRow={false}
    recalculateHiddenLayout={false}
    renderRightHiddenRow={[Function]}
    renderRow={[Function]}
    renderScrollComponent={[Function]}
    rightOpenValue={-75}
    style={
      Array [
        Object {},
        Object {
          "marginLeft": 20,
        },
      ]
    }
    swipeToOpenPercent={50}
  >
    <View />
  </RCTScrollView>
</View>
`;
